ietf/ipr/views.py

```diff
@@ -689,18 +689,41 @@ def search(request):
                 if len(start) == 1:
                     first = start[0]
                     doc = first
-                    docs_related = related_docs(first, reverse_relationship=())
-                    drafts_related = [
-                        d2
-                        for d1 in docs_related
-                        for d2 in related_docs(d1, relationship=())
-                        if d2 is not None
-                    ]
-                    docs = list(set(docs_related + drafts_related))
-                    iprs = iprs_from_docs(docs,states=states)
+                    docs = set([first])
+                    docs.update(
+                        related_docs(
+                            first, relationship=("replaces", "obs"), reverse_relationship=()
+                        )
+                    )
+                    docs.update(
+                        set(
+                            [
+                                draft
+                                for drafts in [
+                                    related_docs(
+                                        d, relationship=(), reverse_relationship=("became_rfc",)
+                                    )
+                                    for d in docs
+                                ]
+                                for draft in drafts
+                            ]
+                        )
+                    )
+                    docs.discard(None)
+                    docs = sorted(
+                        docs,
+                        key=lambda d: (
+                            d.rfc_number if d.rfc_number is not None else 0,
+                            d.became_rfc().rfc_number if d.became_rfc() else 0,
+                        ),
+                        reverse=True,
```

Comment:
I think `0 if d == first else 1` would be safer since it doesn't rely on knowing about the Python internals (and because I think it'd have to be `d != first` because
```python
>>> False < True
True
```

Reply:
> (and because I think it'd have to be `d != first` because
> 
> ```python
> >>> False < True
> True
> ```

We have `reverse=True` as an argument to the `sorted` function above. So, I believe `d == first` is the correct one.

``` python
assert sorted([False, True], reverse=True) == [True, False]
```

Anyway, I'll leave the existing code as-is, leaving the template to handle that logic for now.